develop/api_guc.rst

```diff
@@ -336,30 +339,82 @@ Note that it may be useful to use :code:`error` during development testing, and
+citus.max_adaptive_executor_pool_size (integer)
+***********************************************
 
-Sets the maximum number of simultaneously open files for each server process and defaults to 1000. The real-time executor requires two file descriptors for each shard it sends queries to. Increasing this configuration parameter will allow the executor to have more open file descriptors, and hence handle more shards in parallel. This change has to be made on the workers as well as the coordinator, and can be done only during server start.
+The maximum number of connections per worker node used by the adaptive executor
+to execute a multi-shard command.
 
-.. note::
-  Along with max_files_per_process, one may also have to increase the kernel limit for open file descriptors per process using the ulimit command.
+The adaptive executor opens multiple connections per worker node when running
+multi-shard commands to parallelize the command across multiple cores on the
+worker. This setting specifies the maximum number of connections it will open.
+(The number of connections is also bounded by the number of shards on the
+node.)
 
-citus.enable_repartition_joins (boolean)
-****************************************
+Although increasing parallelism for a single query is generally good, it also
+puts connection pressure on the workers in a production environment.  This GUC
+provides a knob to trade off parallelism vs connection pressure.
 
-Ordinarily, attempting to perform :ref:`repartition_joins` with the real-time executor will fail with an error message. However setting ``citus.enable_repartition_joins`` to true allows Citus to temporarily switch into the task-tracker executor to perform the join. The default value is false.
+The default value is 16, however it's hard to make a good static choice
+manually. In the future the adaptive executor may adjust this GUC on the fly,
+shrinking the pool when workers are busy, and growing it when they are idle.
+
+citus.max_cached_conns_per_worker (integer)
+*******************************************
+
+Each backend opens connections to the workers to query the shards. At the end
+of the transaction, the configured number of connections is kept open to speed
+up subsequent commands.  Increasing this value will reduce the latency of
+multi-shard queries, but will also increase overhead on the workers.
+
+The default value is 1. A larger value such as 2 might be helpful for clusters
+that use a small number of concurrent sessions, but it's not wise to go much
+further (e.g. 16 would be too high).
+
+citus.propagate_set_commands (enum)
+***********************************
+
+Determines which SET commands are propagated from the coordinator to workers.
+The default value for this parameter is 'none'.
+
+The supported values are:
+
+* **none:** no SET commands are propagated.
+
+* **local:** only SET LOCAL commands are propagated.
+
+citus.force_max_query_parallelization (boolean)
```

Comment:
Would it make sense to mention the use case for this GUC we discussed in the meeting?

```SQL
BEGIN;

-- transaction starts with a very lightweight query,
-- which triggers very little connections to be used
-- by the executor
SELECT count(*) FROM table WHERE filter = x;

-- now, we've a very resource intensive query
-- that we want the executor to utilize as much resource/connection
-- as possible
-- however, the executor already decided the resource at the first
-- command in the transaction block
SELECT ... very .. complex..SQL;
````

So, to help with that case:
```SQL
BEGIN;
SET citus.force_max_query_parallelization TO ON;

SELECT count(*) FROM table WHERE filter = x;

SELECT ... very .. complex..SQL;
````
Now, both queries use as much resources as possible although the first command doesn't necessarily need it.